Release: update inverse bond pricing address, UI Updates, react-query for wrap page

appleseed-iii opened the pull request:
UI Updates:
- nav updates for sidebar
- deep links on treasury tabs

Also:
- balancer & curve lp bond support
- grants tab on give
- jones & impermax on wallet

sam-potter committed:
Implement `react-query` for the wrap page (#1515)

* Use common metric components

* Abstracted balances

* Combine switching/adding network to one hook

* Abstract switch network component

* Update test snapshot

* Convert mutations

* Update test snapshots

* Abstract WalletConnectedGuard

* Convert cross chain migration

* Cleanup

* Update test snapshots

* Stop making unecessary requests to testnets

* Add react query error handler

* Update import

* Update test snapshots

* Move file

* Add react-query devtools

* Add error handler

* Add support for migrating specific amounts

* Refactor

* Remove unused migrate functions

* Fix translations

* Use testnets

* Prefer global query error handler

* Fix incorrect boolean

* Use testnets

* Abstract ReactQueryProvider

* Use new DecimalBigNumber helper

* Update test

* Cleanup test

* update snapshot

* Update snapshots

* switch migrate input from label to placeholder

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>
Co-authored-by: brightiron <brightiron@protonmail.com>

biwano committed:
Added missing translation tag in Wrap page (#1612)

dependabot[bot] committed:
build(deps): bump @multifarm/widget from 0.0.108 to 0.0.111 (#1611)

dependabot[bot] committed:
build(deps-dev): bump fast-check from 2.23.0 to 2.23.2 (#1613)

Bumps [fast-check](https://github.com/dubzzz/fast-check) from 2.23.0 to 2.23.2.
- [Release notes](https://github.com/dubzzz/fast-check/releases)
- [Changelog](https://github.com/dubzzz/fast-check/blob/main/CHANGELOG.md)
- [Commits](https://github.com/dubzzz/fast-check/compare/v2.23.0...v2.23.2)

---
updated-dependencies:
- dependency-name: fast-check
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump @ethersproject/address from 5.5.0 to 5.6.0 (#1610)

Bumps [@ethersproject/address](https://github.com/ethers-io/ethers.js/tree/HEAD/packages/address) from 5.5.0 to 5.6.0.
- [Release notes](https://github.com/ethers-io/ethers.js/releases)
- [Changelog](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ethers-io/ethers.js/commits/v5.6.0/packages/address)

---
updated-dependencies:
- dependency-name: "@ethersproject/address"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brightiron committed:
Add Jones Farm Pool and Impermax Borrow (#1617)

* add jones staking pool

* add impermax to borrow

* updating component-library with new assets

* snapshots

* remove console log

dependabot[bot] committed:
build(deps-dev): bump @nomiclabs/hardhat-waffle from 2.0.2 to 2.0.3 (#1615)

Bumps [@nomiclabs/hardhat-waffle](https://github.com/nomiclabs/hardhat) from 2.0.2 to 2.0.3.
- [Release notes](https://github.com/nomiclabs/hardhat/releases)
- [Commits](https://github.com/nomiclabs/hardhat/compare/@nomiclabs/hardhat-waffle@2.0.2...@nomiclabs/hardhat-waffle@2.0.3)

---
updated-dependencies:
- dependency-name: "@nomiclabs/hardhat-waffle"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brightiron committed:
Bond Support for Balancer and Curve LPs  (#1576)

* balancer abis

* refactor v2 bond details to support adding additional contracts

* remove console log

* rename variables

* updating tests

* revert tests

* curve pool support

0xJem committed:
Adds Grants tab to Give (#1489)

* Cleanup of redundant variables

* Remove redundant types and variables

* Fix issue with type definition for donationInfo

* Restore yield image that is incorrect in the component library

* removing svgs

* Fix colour of vault icon

* Tab indicator 4px from tab

* Shift spacing between give tabs and content to 40px

* Align project title with image

* Prevent wrapping of give info titles

* Hopefully a fix for translation issues

* Vertical spacing of cause card content

* Update sOHM goal copy

* Confirmation modal text padding fix

* Fixed edit and withdrawal modal close calls

* Update indication of project name in deposit modal

* Testing a different approach to translation strings

* Another approach to translations

* Trying different approaches to translation

* Translation fix and update README

* Linting fixes

* Constrain give info component on smaller screens

* Top-align project information on medium breakpoint

* Constrain project image height

* Consistent 30px spacing on project card

* Vertical alignment on mobile

* 30px margin on mobile view

* Better handling of breakpoints

* Cleaner className when checking for screen size

* Fix give info on large screens

* Updating image sizing for mobile and tablet

* WIP project info re-org

* Manage Donation modal re-titling

* Cause profile image size and info alignment

* Added new Your Donations paper

* Manage donation modal padding

* Abbreviate text to keep things on one line

* Fix padding on CTA button

* Adjust bottom padding for project details

* Fix image sizing

* Consistency in padding

* Fix fill colour for the give sOHM icon

* Fix issue with formatted string

* Redeem tab-panel view desktop, tablet, mobile

* Slight tweaks to mobile view, updated modal

* Adjust spacing to meet spec

* Adjust spacing according to spec

* Fixing Causes and Redeem padding

* cleaned up class naming for Redeem a bit

* Manage donation modal respects breakpoints

* Fixed checking if wallet address is valid

* Remove console.logs

* Reset projects.json to have correct wallets

* Update cause profile image height

* Added useLocation to DepositRow to allow mock_sohm to work

* Moving as many things to one line as possible for My Donations

* Don't spew "false" into element classes

* TS refactor

* Resolve lint warnings

* Misc CSS cleanup

* Adjust donations table to match spec

* Adjust spacing of education card components (according to spec)

* Correct padding on my donations screen

* Re-implement give routes. Also support URL re-writing when selecting tabs.

* Simplify some code

* Update snapshots with recent changes

* Light update to comments

* Remove edit functionality from RecipientModal (shifted to ManageDonationModal)

* Refactor RecipientModal to be easier to understand

* Better commenting on slice and helper logic

* Fixing non-handling of error on network changes

* Add grants cohort 0 info, images, and create new tab

* Render Grants cards and details pages

* Shift buttons in RecipientModal to component-library

* Project recipient field

* Shift amount field in RecipientModal to the component library

* Remove unused import

* Shift recipient address to component librar

* Replace SvgIcon

* Fix alignment of the education graphics

* RecipientModal: migrate to Modal from component-library

* Update jest snapshot

* Fix rendering bug with modal on Firefox

* Shift to component-library's Modal

* Cleanup

* Shifted ManageDonationModal to be less complicated

* Migrate buttons

* Adjust spacing for ArrowGraphic

* Migrate ManageDonationModal to component-library. Fix build warnings.

* Update images, change button sizing

* Correct implementation of modal onClose and back button

* Migrate RedeemYieldModal to component-library

* Migrate RedeemYield to use component-library

* Remove redundant view

* Migrate ProjectCard to component-library

* Migrate YieldRecipients to component-library

* Migrate GiveInfo

* Partial migration of Give to component-library

* Migrate Give to component-library

* Migrate DepositRow to component-library

* Partial migration of Causes Dashboard to component-library

* Workaround for component-library issue

* Update jest snapshots

* Fix failing test

* Reworked styling, updated short description copy, and added grants explainer

* Partial fixes for spacing in project details

* Cleanup spacing in project details

* Fix positioning of edit button

* Adjust spacing for countdown timer

* Fix right-alignment of sOHM Yield Sent

* move this one component back to muipaper

* Remove obsolete Typography components, update to component-library 1.7.0

* Define schema for milestones and shift GrantsDashboard to use it

* WIP milestones in grants card

* Fix failing test due to changes

* Compilation fix

* Shift GrantCard to component-library

* WIP milestone progress indicator

* Return to grants page

* WIP implementation of the milestone progress bar (using an external library)

* Complete spacing and styling of milestone progress bar, plus milestone details.

* Handle out of bonds index

* Milestone amount

* WIP migration of the grant data section to a responsive Grid

* Bold

* Working on icon updates

* Complete layout of grant data

* Capitalisation

* Fix displaying donor count

* Remove extraneous log

* Fix button position on edit donation card

* Shift Grant screen to use Grid instead of hacky CSS

* Remove extraneous classes

* About section layout

* Fix positioning on grants profile

* Correct sizing of grant image

* Shift more elements to Grid

* Shift donations section to Grid

* Shift to using the topLeft prop of the Paper component and Grid

* Fix sizing of grant image in medium breakpoint

* Correct sizing of milestone details

* Add new icons (though they have theme problems)

* Correct sizing of SVG icons

* Update failing snapshot test

* Remove redundant code

* Align grant profile with design 2.0.2

* Add the ability to hide a grant from display

* Update grants milestone amounts

* Fix rendering of the donors icon with themes. This required explicitly adding a fill property, and defining the stroke property as currentcolor

* Workaround for SVG fills

* Add remaining 2 SVG icons for Grants (with stroke/fill fixes)

* Fix wrapping of donors

* Remove redundant grant

* Didn't mean to enable that

* Correct layout on md and lg breakpoints

* Fix SVG rendering of fills

* SVG updates
- Add new/updated icons
- Remove redundant icons
- Fix SVG background/fill issues

These SVG icons will be moved into component-library shortly.

* Update EducationCard with latest SVG icons

* Hide project stats for grants

* Remove extraneous log

* Update to component-library 1.7.1

* Shift icons in GrantCard to component-library

* Migrate ProjectCard to component-library icons

* Shift EducationCard to use component-library icons

* Remove redundant SVG files

* Add feature flag to disable Give Grants

* Misc design fixes

* Update snapshots

* Disable individual grant projects until they are approved. Also display helpful text if no grants are active.

* Redundant Grid component

* Revert feature flag for give grants (since we will display an informative message instead)

* Compilation fixes due to merge

* Remove redundant CSS

* Shift CauseCard to use Grid component

* WIP refactoring of ProjectCard to use Grid component

* Remove redundant className

* Remove redundancies

* Revert temporary change to user donating flag

* Remove redundant CSS

* Remove redundant CSS

* Update test snapshot with recent changes

* Remove redundant CSS

* Simplify the image used in Project and Grant screens

* Convert amount screen of RecipientModal to use the Grid component

* WIP conversion of RecipientModal to Grid component

* Fix breakpoints on confirmation screen.

* Align RecipientModal with reference design

* Migrate GiveInfo. Remove redundant CSS.

* Update snapshot

* Shift CausesDashboard to use MUI components

* Cleanup Give component

* Update snapshot

* Migrate CausesDashboard and GrantsDashboard

* Clean up redundant CSS (LOTS of it)

* Fix breakpoints for project and grant card

* Responsive version of no donations message

* Migrate manage donation screen

* Migrate ManageDonationModal

* Shift to use GiveBox

* Convert RedeemYield to use Grid. Remove redundant CSS.

* Use the DataRow component

* Remove redundant component

* Adjust breakpoint

* Accidental commit

* Migrate RedeemYieldModal to use Grid. Cleanup CSS.

* Rebase fix

* Layout fixes. Remove custom styling where possible.

* Display message if there is no milestone information

* Clean up EducationCard: shift to Grid components, remove redundant graphics, rename components.

* Remove redundant SVG fill

* Remove hard-coded temporary value

* Center-align the right-facing arrow

* Backport text alignment

* Update snapshot

* Fix Paper width. Remove extraneous padding around causes/grant dashboards.

* Apply fullWidth to Paper

* Better handling of the main Give container at different breakpoints

* Update snapshots

* Enable thunks to record the grant name instead of a wallet address

* Fix bug introduced in a recent version of component-library

* Back-port alignment fix from 8856ace4457429c398c2478760b30ba72329fd18

* Migrate YieldRecipients to use Grid

* Fix an issue with the selected tab being donations

Co-authored-by: brightiron <brightiron@protonmail.com>
Co-authored-by: Lienid <0xLienid@protonmail.com>
Co-authored-by: 0xLienid <92545857+0xLienid@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.9.0 (#1467)

* Bump eslint from 7.32.0 to 8.9.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.9.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.9.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump eslint from 7.32.0 to 8.9.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.9.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.9.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: resolve eslint dep error

Signed-off-by: ivelin <ivelin@ambianic.ai>

* build: fix webpack 5 util polyfill error

Signed-off-by: ivelin <ivelin@ambianic.ai>

* build: cleanup deps

Signed-off-by: ivelin <ivelin@ambianic.ai>

* build: cleanup deps

Signed-off-by: ivelin <ivelin@ambianic.ai>

* chore: address jem's comments:

Signed-off-by: ivelin <ivelin@ambianic.ai>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: ivelin <ivelin@ambianic.ai>

biwano committed:
Adds swedish as a translable language (#1627)

Co-authored-by: Bruno Ilponse <bruno.ilponse@gmail.com>

dependabot[bot] committed:
Bump ethers from 5.5.4 to 5.6.2 (#1625)

Bumps [ethers](https://github.com/ethers-io/ethers.js/tree/HEAD/packages/ethers) from 5.5.4 to 5.6.2.
- [Release notes](https://github.com/ethers-io/ethers.js/releases)
- [Changelog](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ethers-io/ethers.js/commits/v5.6.2/packages/ethers)

---
updated-dependencies:
- dependency-name: ethers
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump hardhat from 2.9.1 to 2.9.2 (#1619)

Bumps [hardhat](https://github.com/nomiclabs/hardhat) from 2.9.1 to 2.9.2.
- [Release notes](https://github.com/nomiclabs/hardhat/releases)
- [Commits](https://github.com/nomiclabs/hardhat/compare/hardhat@2.9.1...hardhat@2.9.2)

---
updated-dependencies:
- dependency-name: hardhat
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

0xJem committed:
Add additional convenience methods to DecimalBigNumber (#1606)

appleseed-iii committed:
remove token addresses from networkDetails.ts so that we only use token addresses in addresses.ts (#1633)

- add comments in addresses.ts for some new rinkeby contracts that are partially migrated
  - the new rinkeby addresses will sync up ohm v1, ohm v2 & bond_depo to all use the same Assets
- remove unused `getGohmBalFromSohm` from `helpers`
- remove unused `fetchBalances`

brightiron committed:
Console log cleanups (#1623)

* resolve console log error for tabs on treasury dashboard

* resolve invalid props passing on topbar

* removing unused line

* fixing gTheme

* removing unused variables

* removing console log and unused var

* unused var

* snapshots

dylan5394 committed:
Added unit test for staking approval button interaction (#1569)

* Increased staking coverage

* updated submodule

0xMakka committed:
add deep links to treasury tabs (#1622)

* add deep links to treasury tabs

* fix failing snapshot

* add new routes for tabs, update url on click of each treasury tab

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

brightiron committed:
Unit Test Improvements (#1632)

brightiron committed:
Nav updates (#1624)

* add jones staking pool

* add impermax to borrow

* sidebar refactor to design system

* nav update to v2 design system

* update component-library

* snapshots

* increasing width

* social link colors

* removing new chip from give

* snapshots

* upgrading component-library

* updating snapshots

* update component library

* component library

* component-library release

* snapshot updates

* update snapshot

* snapshots

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

biwano committed:
I18n/ru and pl (#1616)

* Added missing translation tag in Wrap page

* Adding russian and polish translations to ui

Co-authored-by: Bruno Ilponse <bruno.ilponse@gmail.com>
Co-authored-by: appleseed-iii <80423742+appleseed-iii@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin (#1636)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.12.1 to 5.18.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.18.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sam-potter committed:
Update contract address (#1642)

0xJem committed:
Give: Improve consistency and accuracy of numbers displayed (#1562)

* Migrate ProjectCard to component-library

* Migrate YieldRecipients to component-library

* Migrate GiveInfo

* Partial migration of Give to component-library

* Migrate Give to component-library

* Migrate DepositRow to component-library

* Partial migration of Causes Dashboard to component-library

* Workaround for component-library issue

* Update jest snapshots

* Fix failing test

* Reworked styling, updated short description copy, and added grants explainer

* Partial fixes for spacing in project details

* Cleanup spacing in project details

* Fix positioning of edit button

* Adjust spacing for countdown timer

* Fix right-alignment of sOHM Yield Sent

* move this one component back to muipaper

* Remove obsolete Typography components, update to component-library 1.7.0

* Define schema for milestones and shift GrantsDashboard to use it

* WIP milestones in grants card

* Fix failing test due to changes

* Compilation fix

* Shift GrantCard to component-library

* WIP milestone progress indicator

* Return to grants page

* WIP implementation of the milestone progress bar (using an external library)

* Complete spacing and styling of milestone progress bar, plus milestone details.

* Handle out of bonds index

* Milestone amount

* WIP migration of the grant data section to a responsive Grid

* Bold

* Working on icon updates

* Complete layout of grant data

* Capitalisation

* Fix displaying donor count

* Remove extraneous log

* Fix button position on edit donation card

* Shift Grant screen to use Grid instead of hacky CSS

* Remove extraneous classes

* About section layout

* Fix positioning on grants profile

* Correct sizing of grant image

* Shift more elements to Grid

* Shift donations section to Grid

* Shift to using the topLeft prop of the Paper component and Grid

* Fix sizing of grant image in medium breakpoint

* Correct sizing of milestone details

* Add new icons (though they have theme problems)

* Correct sizing of SVG icons

* Update failing snapshot test

* Remove redundant code

* Align grant profile with design 2.0.2

* Add the ability to hide a grant from display

* Update grants milestone amounts

* Fix rendering of the donors icon with themes. This required explicitly adding a fill property, and defining the stroke property as currentcolor

* Workaround for SVG fills

* Add remaining 2 SVG icons for Grants (with stroke/fill fixes)

* Fix wrapping of donors

* Remove redundant grant

* Didn't mean to enable that

* Correct layout on md and lg breakpoints

* Fix SVG rendering of fills

* SVG updates
- Add new/updated icons
- Remove redundant icons
- Fix SVG background/fill issues

These SVG icons will be moved into component-library shortly.

* Update EducationCard with latest SVG icons

* Hide project stats for grants

* Remove extraneous log

* Update to component-library 1.7.1

* Shift icons in GrantCard to component-library

* Migrate ProjectCard to component-library icons

* Shift EducationCard to use component-library icons

* Remove redundant SVG files

* Add feature flag to disable Give Grants

* Misc design fixes

* Update snapshots

* Disable individual grant projects until they are approved. Also display helpful text if no grants are active.

* Redundant Grid component

* Revert feature flag for give grants (since we will display an informative message instead)

* Compilation fixes due to merge

* Remove redundant CSS

* Shift CauseCard to use Grid component

* WIP refactoring of ProjectCard to use Grid component

* Remove redundant className

* Remove redundancies

* Revert temporary change to user donating flag

* Remove redundant CSS

* Remove redundant CSS

* Update test snapshot with recent changes

* Remove redundant CSS

* Simplify the image used in Project and Grant screens

* Convert amount screen of RecipientModal to use the Grid component

* WIP conversion of RecipientModal to Grid component

* Fix breakpoints on confirmation screen.

* Align RecipientModal with reference design

* Migrate GiveInfo. Remove redundant CSS.

* Update snapshot

* Shift CausesDashboard to use MUI components

* Cleanup Give component

* Update snapshot

* Migrate CausesDashboard and GrantsDashboard

* Clean up redundant CSS (LOTS of it)

* Fix breakpoints for project and grant card

* Responsive version of no donations message

* Migrate manage donation screen

* Migrate ManageDonationModal

* Shift to use GiveBox

* Convert RedeemYield to use Grid. Remove redundant CSS.

* Use the DataRow component

* Remove redundant component

* Adjust breakpoint

* Accidental commit

* Migrate RedeemYieldModal to use Grid. Cleanup CSS.

* Rebase fix

* Layout fixes. Remove custom styling where possible.

* Display message if there is no milestone information

* Clean up EducationCard: shift to Grid components, remove redundant graphics, rename components.

* Remove redundant SVG fill

* Improve precision of give deposits, and indicate when a number has been rounded

* More cleanup for precision

- Standardise on 2 decimal places
- Flag for when deposit data is loading
- Remove use of redundant parseFloat

* Precision fixes for ProjectCard

- Use toFormat instead of toFixed
- Consistent formatting between project and deposit data

* Precision fixes for GrantCard

* Remove hard-coded temporary value

* Precision fixes for RecipientModal

* Fixed precision and formatting. Also fixed calculation of the goal achievement.

* Center-align the right-facing arrow

* Fix issue with text alignment on the redeem screen. Precision fixes.

* Backport text alignment

* Precision fixes for RedeemYieldModal

* Fix incorrect numbers on redeem page. Precision accuracy.

* Accuracy for the donations table

* Update snapshot

* Update snapshot

* Fix Paper width. Remove extraneous padding around causes/grant dashboards.

* Apply fullWidth to Paper

* Better handling of the main Give container at different breakpoints

* Update snapshots

* Multiply, then divide

* Less-confusing approximation symbol

* Fix alignment on yield recipients screen

* Update snapshot

* Enable thunks to record the grant name instead of a wallet address

* Add support for initialising BigNumber with a number

* WIP conversion of Give to DecimalBigNumber

* WIP shift of donation modals

* Add copy function

* WIP conversion

* Add .eq and .lt

* Use .eq

* Formatted string, not accurate string

* Complete ManageDonationModal

* Remove bignumber.js as a dependency

* ProjectCard done

* Remove remaining usage of BigNumber

* Change toFormattedString to default to the DecimalBigNumber's configured decimal places, instead of 0

* Correct tests

* Add trimming of trailing zeroes

* Trim trailing zeroes on ProjectCard

* Test for trailing decimal point and zeroes in formatted strings

* Fix bug introduced in a recent version of component-library

* Fix user input bugs with RecipientModal, exacerbated by the shift to DecimalBigNumber

* Simplify string formatting

* Fix duplicate key issue

* Revert changes

* Remove redundant log

* Unique key in ProjectCard

* Unique key in GrantCard

* Better handling of number formatting

* Number trimming in DepositRow

* Cleanup errors and string formatting on ManageDonationModal

* Apply number trimming on remaining screens

* Make `trimTrailingZeroes` an optional arg. Default formatting to 0

* Document design decisions, add tests

* Add test cases

* Adjust to refinements to DecimalBigNumber

* Add snapshot for GrantCard

* Add snapshot for ProjectCard

* Right align manage button

* Make the decimals parameter optional and determine the number of decimal places

* WIP implementation of consolidated toString() function

* Implemented omitting and padding of decimals

* Add number formatting. ALL GREEN.

* Add test cases for toApproxNumber()

* Replace DecimalBigNumber usage of toAccurateString()

* Remove toAccurateString()

* Remove toFormattedString()

* Update test snapshots

* Update test snapshots

* Shift to new DecimalBigNumber functions

* Fix undefined error

* Add constructor overloads

* Parse input before string inference

* Make args default to `{}`

* Prefer ethers `commify` method

* Fix issue with loading goal completion

* Add default format for decimal places

* Refactor decimal parsing implementation

* Slight implementation changes

* Make decimal values optional

* Default to 0 rather than throw

* Added `_ensurePositive` method

* Update test snapshot (and avoid failing tests)

* Refactor

* mul/div test cases without decimals, and fix an edge case

* Remove redundant specification of decimal places

* Maintain precision when multiplying

* Add more comprehensive tests on division, and a note about precision

* Fix loading dependencies

* Missing space

* Simplify formatting

* Fix tests

- Added testid attributes
- Update snapshots with changes to accuracy

Co-authored-by: Lienid <0xLienid@protonmail.com>
Co-authored-by: brightiron <brightiron@protonmail.com>
Co-authored-by: Sam Potter <15123145+sam-potter@users.noreply.github.com>

brightiron committed:
More Unit Test Improvements  (#1638)

* token approval test

* adding wallet tests

* removing console log

* removing console log

* resolving a failing test case

* adding missing package

* farm pool tvl tests

dependabot[bot] committed:
build(deps-dev): bump @types/lodash from 4.14.179 to 4.14.181 (#1641)

Bumps [@types/lodash](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/lodash) from 4.14.179 to 4.14.181.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/lodash)

---
updated-dependencies:
- dependency-name: "@types/lodash"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/runtime from 7.17.2 to 7.17.9 (#1643)

Bumps [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) from 7.17.2 to 7.17.9.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.17.9/packages/babel-runtime)

---
updated-dependencies:
- dependency-name: "@babel/runtime"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brightiron committed:
goback instead of go to stake page (#1639)

biwano committed:
Translation tags for treasurydashboard tab switcher (#1646)

Co-authored-by: Bruno Ilponse <bruno.ilponse@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-react-hooks from 4.3.0 to 4.4.0 (#1631)

Bumps [eslint-plugin-react-hooks](https://github.com/facebook/react/tree/HEAD/packages/eslint-plugin-react-hooks) from 4.3.0 to 4.4.0.
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/HEAD/packages/eslint-plugin-react-hooks)

---
updated-dependencies:
- dependency-name: eslint-plugin-react-hooks
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brightiron committed:
More Unit Test Improvements (#1651)

* token approval test

* adding wallet tests

* removing console log

* removing console log

* resolving a failing test case

* adding missing package

* farm pool tvl tests

* treasury dashboard tests

* wrap tests

Commit:
chore: remove translations submodule (#1650)

* chore: remove transaltions module

Signed-off-by: ivelin <ivelin@ambianic.ai>

* chore: update yarn scripts and gitignore

Signed-off-by: ivelin <ivelin@ambianic.ai>

* fix: checkout translations on build

Signed-off-by: ivelin <ivelin@ambianic.ai>

* fix: translations build script

Signed-off-by: ivelin <ivelin@ambianic.ai>

* fix: translations build script

Signed-off-by: ivelin <ivelin@ambianic.ai>

dependabot[bot] committed:
build(deps): bump typescript from 4.6.2 to 4.6.3 (#1648)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 4.6.2 to 4.6.3.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v4.6.2...v4.6.3)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 12.3.5 to 12.3.7 (#1649)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 12.3.5 to 12.3.7.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v12.3.5...v12.3.7)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ohmygirth commented on src/components/Sidebar/NavContent.tsx:
```diff
@@ -82,83 +83,34 @@ const NavContent: React.FC<NavContentProps> = ({ handleDrawerToggle }) => {
+                        key={i}
+                        onClick={handleDrawerToggle}
+                        label={bond.displayName}
+                        chip={`${bond.discount && trim(bond.discount * 100, 2)}%`}
```
Same as above

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/1644/commits/28aef2c712546e6b4bf90cee3d999ad0267833e0

ohmygirth commented on src/components/Sidebar/NavContent.tsx:
```diff
@@ -82,83 +83,34 @@ const NavContent: React.FC<NavContentProps> = ({ handleDrawerToggle }) => {
+                        key={i}
+                        onClick={handleDrawerToggle}
+                        label={bond.displayName}
+                        chip={`${bond.discount && trim(bond.discount * 100, 2)}%`}
```
Can this not be a chip? Chips should be used for alerts or filter labels. This is a nit but would rather discount text color be red/green text and no chip.

appleseed-iii replied:
resolved in https://github.com/OlympusDAO/olympus-frontend/pull/1644/commits/28aef2c712546e6b4bf90cee3d999ad0267833e0

ohmygirth reviewed (changes requested):
Small nits but for the most part lgtm

appleseed-iii committed:
add readme for inverse bonds (#1645)

* add readme for inverse bonds

* update snapshot

* revert snapshot

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 5.12.1 to 5.18.0 (#1635)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.12.1 to 5.18.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.18.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brightiron committed:
track location wallet button was clicked from (#1662)

brightiron committed:
update getohm links to mainnet ohm address (#1659)

brightiron committed:
adding support for balancer pool (#1661)

brightiron committed:
More Unit Test Improvements (#1655)

* wallet transaction history tests

* stake area hook tests

* v1 bond snapshot

* better wallet test coverage

* zap mobile res test

* additional zap test

GoldenRatio3 committed:
fixed missing loading state on stake APY metrics (#1656)

brightiron committed:
adjusting bond subnav items (#1658)

* adjusting bond subnav items

* fix sortedInverseBonds, remove Highest Discount text

* update snapshots from #1656

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-config-prettier from 8.4.0 to 8.5.0 (#1652)

Bumps [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) from 8.4.0 to 8.5.0.
- [Release notes](https://github.com/prettier/eslint-config-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v8.4.0...v8.5.0)

---
updated-dependencies:
- dependency-name: eslint-config-prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

appleseed-iii commented:
from 27% to 37% branch coverage over the course of this pr. bing bong. lot of good work from the team in here.

appleseed-iii committed:
add --silent flag to suppress warnings in ci logs (#1666)

* add --silent flag to suppress warnings in ci logs

* remove translations
